data/xml/schema.rnc

```diff
@@ -11,26 +11,36 @@ first = element first { text }
 last = element last { text }
 Person = attribute id { xsd:NCName }?, (first? & last)
 
+checksum = attribute hash { xsd:string {minLength="8" maxLength="8" pattern="[0-9a-f]*"} }
+url-with-checksum = element url {
+    xsd:anyURI {pattern="https?://.*"} |
+    (xsd:anyURI {pattern=".*"}, checksum)
+}
+attachment = element attachment {
+    attribute type { xsd:NCName }?,
+    checksum,
+    xsd:anyURI
```

Comment:
This as NCName before, i.e. a non-qualified URI.  Now it can be anything. I don't think it makes sense to allow absolute URIs for attachments.

Reply:
NCName is not allowed to start with a digit; files named after new-style IDs will start with digits.

Reply:
We could probably define an `xsd:string` with a hand-crafted pattern and name it `local-file` or something?

Reply:
That would probably the best way, so scratch my recommendation re /.* I made above :-)

Reply:
Okay, I'll see if I can make that work. Our relative path entries don't start with a slash in the XML, so `/.*` wouldn't currently match anything, which is why your other suggestion confused me. :)